Library/Homebrew/test/dev-cmd/ruby_spec.rb

```diff
@@ -15,3 +15,17 @@
       .and not_to_output.to_stderr
   end
 end
+
+describe "brew ruby -e 'puts \"testball\".f.path'", :integration_test do
```

Comment:
My only reservation is that these integration tests are really slow so I wonder if it's worth adding them for stuff like this?

Reply:
ok, I wasn't aware of this trade-off; I was just thinking more tests are better, but slow tests do hinder development. I haven't seen a pattern of failure with this code, so I suppose we could hold off.

is there any place for a nightly job that runs a more thorough test suite?

Reply:
> ok, I wasn't aware of this trade-off being an active consideration; I was just thinking more tests are better, but slow tests do hinder development. I haven't seen a pattern of failure with this code, so I suppose we could hold off.

Cool, thanks.



> is there any place for a nightly job that runs a more thorough test suite?

Potentially but I think the issues would be:
- figuring out what broke it (if anything)
- identifying/avoiding flakiness

Thoughts?

Reply:
> > is there any place for a nightly job that runs a more thorough test suite?
> 
> Potentially but I think the issues would be:
> 
> * figuring out what broke it (if anything)

yeah a nightly would make it hard to figure out what broke something, but if we run the more extensive tests on every push to master then I think that would provide traceability of failures without slowing down CI in pull requests

> * identifying/avoiding flakiness

flakiness is really hard and I see two ways to address it:

1. try to ignore it by re-running failed tests in CI for pull requests and if they pass a 2nd time ignore them as flaky (pretty much what we do now). a drawback is that it doesn't catch new flakiness very well.

2. try to hunt for flaky tests by running a nightly job that repeats tests N times or until failure (ctest has `--repeat-until-fail N` for this). This requires parsing the test results to see statistics about test failures, but it could at least show some trends for traceability and could hopefully flag the worst offenders.

Reply:
> yeah a nightly would make it hard to figure out what broke something, but if we run the more extensive tests on every push to master then I think that would provide traceability of failures without slowing down CI in pull requests

It'll still slow things down, unfortunately, because we have a maximum number of macOS workers. If this test was Linux only, though, I think it might be worth adding unconditionally?

> * try to ignore it by re-running failed tests in CI for pull requests and if they pass a 2nd time ignore them as flaky (pretty much what we do now). a drawback is that it doesn't catch new flakiness very well.

Yup 😭

Reply:
> If this test was Linux only, though, I think it might be worth adding unconditionally?

yes, it would have caught this most recent regression on Linux, so I think that is valuable. I've added `:needs_linux` in 7455d53f88cd8214266f638dee2f61baa830d367; let me know if that's what you had in mind